### PR TITLE
Feature/json ld converter

### PR DESCRIPTION
**Pull request by ThomasRanvikEriksen**
**Be aware: xml.etree.ElementTree has security issues as library. This will be replaced by defusedxml**

A start

A lot of things i want to do to make it better and more readable.

Main thing is to take most of the code in main.py and make it as a class or function.
Also want to make changes to Full Model and context part

Future wish: Use xsd and/or json schema instead of python dict as configs. @Sveino mentioned to use rdf shcema in the furture

Should mRID exist along side @id? (Decision: Yes - It is included)

Shall we add:
- "cim:Conductor.length": {"qudt:symbol": "km", "@value": 50.0, "@type": "xsd:float", "@id": "http://qudt.org/1.1/vocab/unit#KiloMeter"}
- Add {"@value": "name and description", "@language": "en"} to name, aliasName and description keys?

- See comments

@Sveino 
@geiabu 
@hemaglu 
@Joahimmehl 
@EmilieSkog 
@VetleNormanAE

### CODE REVIEW

File: Python/cim-convert-tool/configEqCim17.py [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+                'cim:GeneratingUnit.minOperatingP': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ActivePower.value'},
+                'cim:GeneratingUnit.nominalP': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ActivePower.value'},
+                'cim:GeneratingUnit.ratedGrossMaxP': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ActivePower.value'},
+                'cim:GeneratingUnit.ratedNetMaxP': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ActivePower.value'}
+            },
+            'attributes': {
+                'cim:Equipment.EquipmentContainer': {'rdf:resource': {'type': 'string'}, 'list': False}
+            }
+        },
+        'cim:ControlArea': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configEqCim17.py]: Defined in line 136. this should be removed

File: Python/cim-convert-tool/configEqCim17.py [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+                'cim:Equipment.aggregate': {'type': 'bool', 'list': False},
+                # 'cim:Equipment.networkAnalysisEnabled': {'type': 'bool', 'list': False},
+                'cim:Equipment.normallyInService': {'type': 'bool', 'list': False},
+                'cim:EquivalentInjection.maxP': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ActivePower.value'},
+                'cim:EquivalentInjection.maxQ': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ReactivePower.value'},
+                'cim:EquivalentInjection.minP': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ActivePower.value'},
+                'cim:EquivalentInjection.minQ': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ReactivePower.value'},
+                'cim:EquivalentInjection.regulationCapability': {'type': 'bool', 'list': False}
+            },
+            'attributes': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configEqCim17.py]: Reference to EquipmentContainer missing.

File: Python/cim-convert-tool/configEqCim17.py [hunk 3]
```diff
@@ -0,0 +1,602 @@
...
+                'cim:IdentifiedObject.name': {'type': 'string', 'list': False},
+                'cim:Equipment.aggregate': {'type': 'bool', 'list': False},
+                # 'cim:Equipment.networkAnalysisEnabled': {'type': 'bool', 'list': False},
+                'cim:Equipment.normallyInService': {'type': 'bool', 'list': False},
+                'cim:EnergyConsumer.pfixed': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ActivePower.value'},
+                'cim:EnergyConsumer.pfixedPct': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ActivePower.value'},
+                'cim:EnergyConsumer.qfixed': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ReactivePower.value'},
+                'cim:EnergyConsumer.qfixedPct': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:ReactivePower.value'}
+            },
+            'attributes': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configEqCim17.py]: Reference to LoadResponse not included

File: Python/cim-convert-tool/configEqCim17.py [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+                'cim:IdentifiedObject.mRID': {'type': 'string', 'list': False},
+                # 'cim:IdentifiedObject.aliasName': {'type': 'string', 'list': False},
+                'cim:IdentifiedObject.description': {'type': 'string', 'list': False},
+                'cim:IdentifiedObject.name': {'type': 'string', 'list': False}
+            },
+            'attributes': {
+                'cim:OperationalLimitSet.Terminal': {'rdf:resource': {'type': 'string'}, 'list': False}
+            }
+        },
+        'cim:OperationalLimitSet': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configEqCim17.py]: Should be OperationalLimitType

File: Python/cim-convert-tool/configEqCim17.py [hunk 5]
```diff
@@ -0,0 +1,602 @@
...
+            },
+            'tags': {
+                'cim:IdentifiedObject.mRID': {'type': 'string', 'list': False},
+                # 'cim:IdentifiedObject.aliasName': {'type': 'string', 'list': False},
+                'cim:IdentifiedObject.description': {'type': 'string', 'list': False},
+                'cim:IdentifiedObject.name': {'type': 'string', 'list': False},
+                'cim:OperationalLimitType.acceptableDuration': {'type': 'float', 'list': False, 'CIMDatatype': 'cim:Seconds.value'},
+                'cim:OperationalLimitType.isInfiniteDuration': {'type': 'bool', 'list': False}
+            },
+            'attributes': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configEqCim17.py]: Should include reference to OperationalLimitType.kind and OperationalLimitType.direction.  Reference to terminal is not used.

File: Python/cim-convert-tool/configEqCim17.py [hunk 6]
```diff
@@ -0,0 +1,602 @@
...
+                'cim:Equipment.EquipmentContainer': {'rdf:resource': {'type': 'string'}, 'list': False},
+                'cim:ConductingEquipment.BaseVoltage': {'rdf:resource': {'type': 'string'}, 'list': False}
+            }
+        },
+        'cim:EquivalentInjection': {
+            'mainAttributes': {
+                'rdf:ID': {'type': 'string', 'list': False} # @id
+            },
+            'tags': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configEqCim17.py]: r and x attributes are missing (used in MV_EQ)

**Review by EmilieSkog — CHANGES_REQUESTED** [review]
configEqCim17.py 
The script is missing 

- StationSupply (MV_EQ)
- Line (MV_EQ and LV_EQ)

File: Python/cim-convert-tool/configEqCim17.py (hunk 1 above)
Comment (ThomasRanvikEriksen) [reply Python/cim-convert-tool/configEqCim17.py]: Can you say more?

Comment (EmilieSkog) [reply Python/cim-convert-tool/configEqCim17.py]: Sorry. ControlArea seems to have been defined twice, here in line 262 and above in line 136. The definition in line 136 is the one that corresponds to what we have in our examples.

Comment (ThomasRanvikEriksen) [reply Python/cim-convert-tool/configEqCim17.py]: Fixed

File: Python/cim-convert-tool/configSshCim17.py [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+            },
+            'attributes': {
+
+            }
+        },
+        'cim:ControlArea': {
+            'mainAttributes': {
+                'rdf:about': {'type': 'string', 'list': False} # @id
+            },
+            'tags': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configSshCim17.py]: ControlArea.pTolerance CIM attribute is missing (used in MV file)

File: Python/cim-convert-tool/configSshCim17.py [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+            },
+            'attributes': {
+
+            }
+        },
+        'cim:GeneratingUnit': {
+            'mainAttributes': {
+                'rdf:about': {'type': 'string', 'list': False} # @id
+            },
+            'tags': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configSshCim17.py]: GeneratingUnit.normalPF CIM attribute is missing (used in MV file)

File: Python/cim-convert-tool/configSshCim17.py [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+            },
+            'attributes': {
+
+            }
+        },
+        'cim:LinearShuntCompensator': {
+            'mainAttributes': {
+                'rdf:about': {'type': 'string', 'list': False} # @id
+            },
+            'tags': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configSshCim17.py]: ShuntCompensator.sections and RegulatingCondEq.controlEnabled CIM attributes are missing (used in MV file)

File: Python/cim-convert-tool/configSshCim17.py [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+            },
+            'attributes': {
+
+            }
+        },
+        'cim:SynchronousMachine': {
+            'mainAttributes': {
+                'rdf:about': {'type': 'string', 'list': False} # @id
+            },
+            'tags': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configSshCim17.py]: RotatingMachine.p, RotatingMachine.q, RegulatingCondEq.controlEnabled CIM attributes and SynchronousMachine.operatingMode reference are missing (used in MV file)

**Review by EmilieSkog — CHANGES_REQUESTED** [review]
configSshCim17.py
The script is missing

- RatioTapChanger 
- RegulatingControl 
- TapChangerControl (all in MV_SSH)

File: Python/cim-convert-tool/configReadingTypeCim17.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+                'rdf:ID': {'type': 'string', 'list': False} # @id
+            },
+            'tags': {
+                'cim:IdentifiedObject.mRID': {'type': 'string', 'list': False},
+                # 'cim:IdentifiedObject.aliasName': {'type': 'string', 'list': False},
+                'cim:IdentifiedObject.description': {'type': 'string', 'list': False},
+                'cim:IdentifiedObject.name': {'type': 'string', 'list': False}
+                # 'cim:IdentifiedObject.shortName': {'type': 'string', 'list': False},
+            },
+            'attributes': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configReadingTypeCim17.py]: Missig attribute ReadingType.phases

File: Python/cim-convert-tool/configAcCim17.py [hunk 12]
```diff
@@ -0,0 +1,110 @@
+configAcCim17 = \
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configAcCim17.py]: config script looks fine, but there seems to be some errors in the [DIGIN10-30-M1_AC.xml]( https://github.com/digin-energi/Grunnprofil/blob/develop/DIGIN10/Asset/DIGIN10-30-M1_AC.xml)

Manufacturer.IdentifiedObject.alias missing aliasName?
	
ProductAssetModel keeps attribute as if inherited by AssetModelCatalogueItem, but there is no such inheritance in the UML. Only an association from ProductAssetModel to AssetModelCatalogueItem.

File: Python/cim-convert-tool/configDlCim17.py [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+        'cim:DiagramObject': {
+            'mainAttributes': {
+                'rdf:ID': {'type': 'string', 'list': False} # @id
+            },
+            'tags': {
+                'cim:IdentifiedObject.mRID': {'type': 'string', 'list': False},
+                # 'cim:IdentifiedObject.aliasName': {'type': 'string', 'list': False},
+                'cim:IdentifiedObject.description': {'type': 'string', 'list': False},
+                'cim:IdentifiedObject.name': {'type': 'string', 'list': False},
+                'cim:IdentifiedObject.shortName': {'type': 'string', 'list': False}
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configDlCim17.py]: attribute DiagramObject.rotation missing

File: Python/cim-convert-tool/configDlCim17.py [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+            },
+            'attributes': {
+
+            }
+        },
+        'cim:TextDiagramObject': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configDlCim17.py]: Missing attribute TextDiagramObject.text

File: Python/cim-convert-tool/configOpCim17.py [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+            },
+            'attributes': {
+                'cim:Measurement.phases': {'rdf:resource': {'type': 'string'}, 'list': False},
+                'cim:Measurement.unitMultiplier': {'rdf:resource': {'type': 'string'}, 'list': False},
+                'cim:Measurement.unitSymbol': {'rdf:resource': {'type': 'string'}, 'list': False},
+                'cim:Measurement.Asset': {'rdf:resource': {'type': 'string'}, 'list': False},
+                'cim:Measurement.PowerSystemResources': {'rdf:resource': {'type': 'string'}, 'list': False}
+            }
+        },
+        'cim:AnalogValue': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configOpCim17.py]: AnalogValue missing reference MeasurementValue.MeasurementValueSource and MeasurementValue.RemoteSource

File: Python/cim-convert-tool/configOpCim17.py [hunk 16]
```diff
@@ -0,0 +1,182 @@
+configOpCim17 = \
+    {
+        'cim:Analog': {
```
Comment (EmilieSkog) [inline Python/cim-convert-tool/configOpCim17.py]: Analog missing reference to Measurement.ACDCTerminal

**Review by EmilieSkog — CHANGES_REQUESTED** [review]

**Comment by ThomasRanvikEriksen** [comment]
@Sveino comment for now is only to remove namespace for mRID and only include the uuid.

We will keep types for values for now

When decimal (money) add a float type with only 2 decimals

Else okay

**Pull request merged**